var/spack/repos/builtin/packages/dlib/package.py

```diff
@@ -81,3 +88,14 @@ def cmake_args(self):
             )
 
         return args
+
+    @run_after("build")
+    def build_python(self):
+        if self.spec.satisfies("+python"):
+            pip = self.spec["python"].command
+            pip.add_default_arg("-m", "pip")
+            env = EnvironmentModifications()
+            env.set("CMAKE_BUILD_PARALLEL_LEVEL", str(make_jobs))
+            pip.add_default_envmod(env)
```

Comment:
This seems a bit non-standard. Can we not use `setup_build_environment` for this?

Reply:
It only needs to be set for the pip part of the build.  I guess we could put it in setup_build_envionment